.drone/pipelines/default/runtime/cats_common.sh

```diff
@@ -0,0 +1,191 @@
+    do sleep 1; timeout=$((timeout - 1))
+    done
+    if [[ "${timeout}" == 0 ]]; then
+	# shellcheck disable=SC2005
+	>&2 red "Timed out waiting for the ig job to complete"
+	return 1
+    fi
+
+    # shellcheck disable=SC2005
+    blue "Completed ig job, continue..."
+    return 0
+}
+
+cf_acceptance_tests_job_name() {
+    kubectl get jobs \
+	    --namespace "${KUBECF_NAMESPACE}" \
+	    --output name \
+	    2> /dev/null \
+	| grep "acceptance-tests"
+}
+
+cf_acceptance_tests_pod_name() {
+    kubectl get pods \
+	    --namespace "${KUBECF_NAMESPACE}" \
+	    --output name \
+	    2> /dev/null \
+	| grep "acceptance-tests"
+}
+
+cf_acceptance_tests_pod_exists() {
+    kubectl get pods \
+	    --namespace "${KUBECF_NAMESPACE}" \
+	    --output name \
+	    2> /dev/null \
+	| grep --quiet "acceptance-tests"
+}
+
+cf_acceptance_tests_pod_running() {
+    pod_name="${1}"
+    kubectl get "${pod_name}" \
+	    --namespace "${KUBECF_NAMESPACE}" \
+	    --output jsonpath='{.status.containerStatuses[?(@.name == "acceptance-tests-acceptance-tests")].state.running}' \
+	    2> /dev/null
+}
+
+# Wait for cf-acceptance-tests to start.
+wait_for_cf_acceptance_tests_pod() {
+    # shellcheck disable=SC2005
+    echo "$(blue "Waiting for the cf-acceptance-tests pod to start...")"
+    local timeout="300"
+    until cf_acceptance_tests_pod_exists || [[ "$timeout" == "0" ]]
+    do
+	sleep 1; timeout=$((timeout - 1))
+    done
+    if [[ "${timeout}" == 0 ]]; then
+	# shellcheck disable=SC2005
+	>&2 echo "$(red "Timed out waiting for the cf-acceptance-tests pod to be created")"
+	return 1
+    fi
+    pod_name="$(cf_acceptance_tests_pod_name)"
+    echo "$(green "Pod exists"):  $(blue "${pod_name}")"
+
+    until [[ "$(cf_acceptance_tests_pod_running "${pod_name}")" != "" ]] || [[ "$timeout" == "0" ]]
+    do
+	sleep 1; timeout=$((timeout - 1))
+    done
+    if [[ "${timeout}" == 0 ]]; then
+	# shellcheck disable=SC2005
+	>&2 echo "$(red "Timed out waiting for the cf-acceptance-tests pod to come online")"
+	return 1
+    fi
+    echo "$(green "Pod running"): $(blue "${pod_name}")"
+    return 0
+}
+
+# Follow the logs. If the tests fail, the logs command will also fail.
+cf_acceptance_tests_pod_follow_log() {
+    pod_name="${1}"
+    kubectl logs \
+	    --follow "${pod_name}" \
+	    --namespace "${KUBECF_NAMESPACE}" \
+	    --container acceptance-tests-acceptance-tests
+}
+
+# Wait for the container to terminate and provide the container's exit
+# code to the caller
+wait_for_cf_acceptance_tests_termination() {
+    pod_name="${1}"
+    jsonpath='{.status.containerStatuses[?(@.name == "acceptance-tests-acceptance-tests")].state.terminated.exitCode}'
+    while true; do
+	exit_code="$(kubectl get "${pod_name}" --namespace "${KUBECF_NAMESPACE}" --output "jsonpath=${jsonpath}")"
+	if [[ -n "${exit_code}" ]]; then
+	    break
+	fi
+	sleep 1
+    done
+    echo "${exit_code}"
+}
+
+# Core of CAT execution
+cf_acceptance_tests_go() {
```

Comment:
Maybe call this `run_cf_acceptance_tests`?

Reply:
Accepted.

Reply:
Resolved with commit 97720239811d858b79602230c00ca923b197896d